homeassistant/components/hvv_departures/config_flow.py

```diff
@@ -0,0 +1,249 @@
+
+        if not errors:
+            self.data = user_input
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+            )
+
+        return self.async_show_form(
+            step_id="user", data_schema=SCHEMA_STEP_USER, errors=errors
+        )
+
+    async def async_step_station(self, user_input=None):
+        """Handle the step where the user inputs his/her station."""
+        if user_input is None:
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION
+            )
+
+        errors = {}
+
+        check_name = await self.hub.gti.checkName(
+            {"theName": {"name": user_input[CONF_STATION]}, "maxList": 20}
+        )
+
+        stations = check_name.get("results")
+
+        if not stations:
+            errors["base"] = "no_results"
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+            )
+
+        self.stations = {
+            f"{station.get('name')} ({station.get('type')})": station
+            for station in stations
+        }
+
+        schema = vol.Schema(
+            {vol.Required(CONF_STATION): vol.In(list(self.stations.keys()))}
+        )
+
+        return self.async_show_form(
+            step_id="station_select", data_schema=schema, errors=errors
+        )
+
+    async def async_step_station_select(self, user_input=None):
+        """Handle the step where the user inputs his/her station."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="station_select", data_schema=SCHEMA_STEP_STATION
+            )
+
+        self.data.update({"station": self.stations[user_input[CONF_STATION]]})
+
+        # get station information
+        station_information = await self.hub.gti.stationInformation(
+            {"station": self.data[CONF_STATION]}
+        )
+
+        self.data.update({"stationInformation": station_information})
+
+        title = self.data[CONF_STATION]["name"]
+
+        return self.async_create_entry(title=title, data=self.data)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options flow handler."""
+
+    def __init__(self, config_entry):
+        """Initialize HVV Departures options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.departure_filters = {}
+        self.hub = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if not self.departure_filters:
+
+            departure_list = {}
+
+            session = aiohttp_client.async_get_clientsession(self.hass)
+            self.hub = GTIHub(
```

Comment:
We should try to reuse the existing hub that the sensor is using. Save that in hass.data, then we can use it here in the options flow.